projects/ngx-sub-form/src/lib/ngx-sub-form.component.spec.ts

```diff
@@ -328,6 +328,70 @@ describe(`NgxSubFormComponent`, () => {
       }, 0);
     });
   });
+
+  describe('getFormGroupControlOptions', () => {
+    interface Numbered {
+      numberOne: number;
+      numberTwo: number;
+    }
+
+    class ValidatedSubComponent extends NgxSubFormComponent<Numbered> {
+      protected getFormControls() {
+        return {
+          numberOne: new FormControl(null, Validators.required),
+          numberTwo: new FormControl(null, Validators.required),
+        };
+      }
+
+      public getFormGroupControlOptions(): AbstractControlOptions {
+        return {
+          validators: [
+            formGroup => {
+              const sum = formGroup.value.numberOne + formGroup.value.numberTwo;
+
+              if (sum % 2 !== 0) {
+                return {
+                  sumMustBeEven: sum,
+                };
+              }
+
+              return null;
+            },
+          ],
+        };
+      }
+    }
+
+    let validatedSubComponent: ValidatedSubComponent;
+
+    beforeEach((done: () => void) => {
+      validatedSubComponent = new ValidatedSubComponent();
+
+      // we have to call `updateValueAndValidity` within the constructor in an async way
+      // and here we need to wait for it to run
+      setTimeout(() => {
+        done();
+      }, 0);
+    });
+
+    it('can declare custom form group level validators, to prevent updating the control', (done: () => void) => {
+      const spyOnChange = jasmine.createSpy();
+      validatedSubComponent.registerOnChange(spyOnChange);
+
+      validatedSubComponent.formGroup.patchValue({ numberOne: 1, numberTwo: 2 });
+
+      setTimeout(() => {
+        expect(validatedSubComponent.validate()).toEqual({ formGroup: { sumMustBeEven: 3 } });
+
+        validatedSubComponent.formGroup.patchValue({ numberOne: 2, numberTwo: 2 });
+        setTimeout(() => {
+          expect(spyOnChange).toHaveBeenCalledWith({ numberOne: 2, numberTwo: 2 });
+          expect(validatedSubComponent.validate()).toEqual(null);
+          done();
+        }, 0);
```

Comment:
We should have a test case to make sure that `validatedSubComponent.formGroupErrors` is what we expect

Reply:
Done, not that this _should_ be exactly the same as `.validate()` but it is worth the check